Source/Core/RequestScheduler.js

```diff
@@ -0,0 +1,440 @@
+                return deferRequest(request);
+            }
+        }
+
+        if (RequestScheduler.prioritize && defined(request.type) && !request.defer) {
+            var budget = getBudget(request);
+            if (budget.used >= budget.total) {
+                // Request does not fit in the budget, return undefined
+                handleLeftoverRequest(request);
+                return undefined;
+            }
+            ++budget.used;
+        }
+
+        return startRequest(request);
+    };
+
+    /**
+     * A function that will make a request when an open slot is available. Always returns
+     * a promise, which is suitable for data sources and utility functions.
+     *
+     * @param {String} url The URL to request.
+     * @param {RequestScheduler~RequestFunction} requestFunction The actual function that
+     *        makes the request.
+     * @param {Object} [parameters] Extra parameters to send with the request.
+     * @param {RequestType} [requestType] Type of request. Used for more fine-grained priority sorting.
+     *
+     * @returns {Promise.<Object>} A Promise for the requested data.
+     */
+    RequestScheduler.request = function(url, requestFunction, parameters, requestType) {
+        return RequestScheduler.schedule(new Request({
+            url : url,
+            parameters : parameters,
+            requestFunction : requestFunction,
+            defer : true,
+            type : defaultValue(requestType, RequestType.OTHER)
+        }));
+    };
+
+    function clearStats() {
+        stats.numberOfRequestsThisFrame = 0;
+    }
+
+    function showStats() {
+        if (!RequestScheduler.debugShowStatistics) {
+            return;
+        }
+
+        if (stats.numberOfRequestsThisFrame > 0) {
+            console.log('Number of requests attempted: ' + stats.numberOfRequestsThisFrame);
+        }
+
+        var numberOfActiveRequests = RequestScheduler.maximumRequests - RequestScheduler.getNumberOfAvailableRequests();
+        if (numberOfActiveRequests > 0) {
+            console.log('Number of active requests: ' + numberOfActiveRequests);
+        }
+    }
+
+    /**
+     * Clears the request scheduler before each spec.
+     *
+     * @private
+     */
+    RequestScheduler.clearForSpecs = function() {
+        activeRequestsByServer = {};
+        activeRequests = 0;
+        budgets = [];
+        leftoverRequests = [];
+        deferredRequests = new Queue();
+        stats = {
+            numberOfRequestsThisFrame : 0
+        };
+    };
+
+    /**
+     * Specifies the maximum number of requests that can be simultaneously open to a single server.  If this value is higher than
+     * the number of requests per server actually allowed by the web browser, Cesium's ability to prioritize requests will be adversely
+     * affected.
+     * @type {Number}
+     * @default 6
+     */
+    RequestScheduler.maximumRequestsPerServer = 6;
```

Comment:
How did we choose this number, because that's what Chrome does?  I believe other browsers have different values.  If this number is really important, should we have some browser-specific test code in here to determine a better default?